velox/vector/ComplexVector.h

```diff
@@ -722,6 +722,63 @@ class MapVector : public ArrayVectorBase {
   std::shared_ptr<MapVector> update(
       const folly::Range<DecodedVector*>& others) const;
 
+  struct ToRowVectorOptions {
+    // The list of keys to extract from each map. The type should be the same as
+    // the value type in the map. Should be the same size as 'outputFieldNames'.
+    // Cannot be empty.
+    VectorPtr keysToProject;
+
+    // The names to assign to the corresponding fields in the output RowVector.
+    // Should be the same size as 'keysToProject'.
+    std::vector<std::string> outputFieldNames;
+
+    // When set to false, if the map is null, a key is missing, or the
+    // value is null, the output field will also be null.
+    // If set to true, replaces null output values with type-specific defaults
+    // like for primitive types, they will be set to the c++ default initialized
+    // value, like 0 for INTEGER and empty string for VARCHAR. For Array/Map
+    // they will be set to an empty array or map. Not supported for ROW or
```

Comment:
> Not supported for ROW

We can remove this restriction if we let the caller provide the default value. There is no reason to disallow default value for ROW, I think.

Reply:
That's fine, although, it would require changing the API. Hence, shouldn't be delayed.